### PR TITLE
MF-512 - Change service default ports

### LINKED ISSUE #512: Change service default ports
**FEATURE REQUEST**

All adapters default ports are 8180 for http and 8181 for gRPC.
If someone tries to run couple of those adapters locally on his machine, with default settings, there is a port clash. It would be nice to have default ports different, so it would work without any need to set those ports.
Another thing that would be nice is to have ports split by protocol. Something like the example below:

7000 - users grpc
7001 - things grpc

8000 - ws adapter

9000 - users http
9001 - things http
9002 - http adapter

Please note that this change would affect docker compose configuration as well as run.sh script

### PR DESCRIPTION
**Pull request by SammyOina**
### What does this do?
Change the default ports on services, 7XXX for grpc and 9XXX for http. Default adapter ports remain unchanges

### Which issue(s) does this PR fix/relate to?
Resolves #512 

### List any changes that modify/break current functionality
Updated ports:

| service | HTTP port | GRPC port |
| ---------- | --------------  | --------------- |
| Things | 9000 | |
| Things - Auth | 9001 | 7000 |
| Users | 9002 | |
|Cassandra-reader | 9003 | |
| Cassandra-writer | 9004 | |
| influxdb-reader | 9005   | |
| influxdb-writer | 9006   | |
| mongodb reader | 9007    | |
| mongodb writer | 9008    | |
| postgres-reader | 9009   | |
| postgres-writer | 9010   | |
| timescale-reader | 9011  | |
| timescale-writer | 9012  | |
| bootstrap | 9013         | |
| smpp-notifier | 9014     | |
| smtp-notifier | 9015     | |
| provision | 9016      | |   
| lora | 9017         | |     
| twins | 9018      | |
| certs | 9019 | |
| Auth | 9020 | 7001 |
| HTTP adapter | 80|   |       

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### CODE REVIEW

**Comment by dborovcanin** [comment]
Let's keep the most "important" (the most frequently used) services first:
Auth, Things, Users, (Adapters), Writers/Readers, Bootstrap, Provision, Notifiers, LoRA, Twins.
Other than that - this looks very good.

**Comment by rodneyosodo** [comment]
> ### What does this do?
> Change the default ports on services, 7XXX for grpc and 9XXX for http. Default adapter ports remain unchanges
> 
> ### Which issue(s) does this PR fix/relate to?
> Resolves #512
> 
> ### List any changes that modify/break current functionality
> Updated ports:
> 
> service	HTTP port	GRPC port
> Auth	9000	7000
> Things	9001	
> Things - Auth	9002	7001
> Users	9003	
> Cassandra-reader	9004	
> Cassandra-writer	9005	
> influxdb-reader	9006	
> influxdb-writer	9007	
> mongodb reader	9008	
> mongodb writer	9009	
> postgres-reader	9010	
> postgres-writer	9011	
> timescale-reader	9012	
> timescale-writer	9013	
> bootstrap	9014	
> smpp-notifier	9015	
> smtp-notifier	9016	
> provision	9017	
> lora	9018	
> twins	9019	
> ### Have you included tests for your changes?
> No
> 
> ### Did you document any new/modified functionality?
> No
> 
> ### Notes

Certs ports isn't reflected

**Review by rodneyosodo — COMMENTED** [review]
Remember to update https://github.com/mainflux/docs

**Comment by rodneyosodo** [comment]
@dborovcanin and @SammyOina What about this approach

- `900X` - Core HTTP Services
- `901X` - Addon Services
- `902X` - Readers
- `903X` - Writers

**Comment by dborovcanin** [comment]
> @dborovcanin and @SammyOina What about this approach
> 
> * `900X` - Core HTTP Services
> * `901X` - Addon Services
> * `902X` - Readers
> * `903X` - Writers

I like the idea, but, while it's unlikely that we'll have more than 10 services per "scope", it's probably less confusing and more streamlined to pick the next free port.

**Review by dborovcanin — CHANGES_REQUESTED** [review]
@SammyOina Let's make Users 9000, Things 9001, and make `Auth` the largest 9xxx we have. The reason is #1716 - we'll remove `Auth` as of MF 0.14.

**Comment by SammyOina** [comment]
> > ### What does this do?
> > Change the default ports on services, 7XXX for grpc and 9XXX for http. Default adapter ports remain unchanges
> > ### Which issue(s) does this PR fix/relate to?
> > Resolves #512
> > ### List any changes that modify/break current functionality
> > Updated ports:
> > service	HTTP port	GRPC port
> > Auth	9000	7000
> > Things	9001
> > Things - Auth	9002	7001
> > Users	9003
> > Cassandra-reader	9004
> > Cassandra-writer	9005
> > influxdb-reader	9006
> > influxdb-writer	9007
> > mongodb reader	9008
> > mongodb writer	9009
> > postgres-reader	9010
> > postgres-writer	9011
> > timescale-reader	9012
> > timescale-writer	9013
> > bootstrap	9014
> > smpp-notifier	9015
> > smtp-notifier	9016
> > provision	9017
> > lora	9018
> > twins	9019
> > ### Have you included tests for your changes?
> > No
> > ### Did you document any new/modified functionality?
> > No
> > ### Notes
> 
> Certs ports isn't reflected

certs port was originally unaltered, I'll change it in a new commit

**Review by dborovcanin — DISMISSED** [review]

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
Change also docs inside the repo for example `MF_THINGS_AUTH_GRPC_URL` 
- certs/README.md
- bootstrap/README.md
- cmd/provision/main.go L38, L39. L40
- coap/README.md
- consumers/notifiers/smtp/README.md
- http/README.md
- metrics/prometheus.yml
- mqtt/README.md
- pkg/transformers/json/README.md
- provision/README.md
- provision/configs/config.toml
- readers/*/README.md
- things/README.md
- things/api/auth/grpc/setup_test.go 
- tools/provision/provision.go - ReaderURL
- twins/README.md - MF_AUTH_GRPC_URL
- ws/README.md - MF_THINGS_AUTH_GRPC_URL

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
Also change:
- auth/api/grpc/endpoint_test.go L27
- metrics/prometheus.yml L9 - This should be core features plus adapters

**Review by dborovcanin — DISMISSED** [review]

File: metrics/prometheus.yml [hunk 1]
```diff
@@ -6,4 +6,4 @@ scrape_configs:
   - job_name: 'mainflux'
 
     static_configs:
-    - targets: ['localhost:8180', 'localhost:8182', 'localhost:8184', 'localhost:8185', 'localhost:8186', 'localhost:8187']
+    - targets: ['localhost:9002', 'localhost:9000', 'localhost:8080', 'localhost:8185', 'localhost:8186', 'localhost:9017', "localhost:5683", "localhost:8188"]
```
Comment (rodneyosodo) [inline metrics/prometheus.yml]: I have tested with this config and It works. I would suggest

```suggestion
    - targets: ['mainflux-things:9000',  'mainflux-users:9002', 'mainflux-auth:9020', 'mainflux-http:8185', 'mainflux-ws:8186', "mainflux-coap:5683"]
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
Fix CI

**Review by rodneyosodo — DISMISSED** [review]

**Review by rodneyosodo — DISMISSED** [review]

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Let's move the HTTP adapter to the standard port for HTTP protocol (80).

File: metrics/prometheus.yml [hunk 2]
```diff
@@ -6,4 +6,4 @@ scrape_configs:
   - job_name: 'mainflux'
 
     static_configs:
-    - targets: ['localhost:8180', 'localhost:8182', 'localhost:8184', 'localhost:8185', 'localhost:8186', 'localhost:8187']
+    - targets: ['mainflux-things:9000',  'mainflux-users:9002', 'mainflux-auth:9020', 'mainflux-http:8185', 'mainflux-ws:8186','mainflux-coap:5683']
```
Comment (rodneyosodo) [inline metrics/prometheus.yml]: @SammyOina also change here

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: http/README.md [hunk 3]
```diff
@@ -11,12 +11,12 @@ default values.
 | Variable                    | Description                                                   | Default               |
 | --------------------------- | ------------------------------------------------------------- | --------------------- |
 | MF_HTTP_ADAPTER_LOG_LEVEL   | Log level for the HTTP Adapter                                | info                  |
-| MF_HTTP_ADAPTER_PORT        | Service HTTP port                                             | 8180                  |
+| MF_HTTP_ADAPTER_PORT        | Service HTTP port                                             | 8008                  |
```
Comment (dborovcanin) [inline http/README.md]: The default port is `80`.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**